Keep saving optionsfile to home dir, even if there is not one when reading
config for the first time.

Antonio Ospite <ospite@studenti.unina.it>

diff -bpruN uae-0.8.25/src/main.c uae-0.8.25_patched/src/main.c
--- uae-0.8.25/src/main.c	2006-06-07 17:41:49.000000000 +0200
+++ uae-0.8.25_patched/src/main.c	2006-06-08 20:33:05.000000000 +0200
@@ -357,6 +357,7 @@ void parse_cmdline (int argc, char **arg
 static void parse_cmdline_and_init_file (int argc, char **argv)
 {
     char *home;
+    char *pwd_optionsfile;
 
     strcpy (optionsfile, "");
 
@@ -374,8 +375,9 @@ static void parse_cmdline_and_init_file 
     if (! cfgfile_load (&currprefs, optionsfile)) {
 #ifdef OPTIONS_IN_HOME
 	/* sam: if not found in $HOME then look in current directory */
-	strcpy (optionsfile, OPTIONSFILENAME);
-	cfgfile_load (&currprefs, optionsfile);
+  /* ao: but let's use a temporary name, so we save to home dir, yet :) */
+	strcpy (pwd_optionsfile, OPTIONSFILENAME);
+	cfgfile_load (&currprefs, pwd_optionsfile);
 #endif
     }
 
